lib/SILOptimizer/Utils/InstructionDeleter.cpp

```diff
@@ -23,6 +23,8 @@ static bool hasOnlyIncidentalUses(SILInstruction *inst,
   for (SILValue result : inst->getResults()) {
     for (Operand *use : result->getUses()) {
       SILInstruction *user = use->getUser();
+      if (isa<DeallocStackInst>(user) && isa<AllocStackInst>(inst))
```

Comment:
Deleting an `alloc_stack` is supposed to be handled by `isScopeAffectingInstructionDead`. We cannot remove `alloc_stack` at -Onone without checking varInfo